src/config.rs

```diff
@@ -1027,22 +1027,22 @@ impl<'a> Cfg<'a> {
     }
 }
 
+/// The root path of the release server, without the `/dist` suffix.
+/// By default, it points to [`dist::DEFAULT_DIST_SERVER`].
 pub(crate) fn dist_root_server(process: &Process) -> Result<String> {
-    Ok(match non_empty_env_var("RUSTUP_DIST_SERVER", process)? {
-        Some(s) => {
+    Ok(
+        if let Some(s) = non_empty_env_var("RUSTUP_DIST_SERVER", process)? {
             trace!("`RUSTUP_DIST_SERVER` has been set to `{s}`");
             s
         }
-        None => {
-            // For backward compatibility
-            non_empty_env_var("RUSTUP_DIST_ROOT", process)?
-                .inspect(|url| trace!("`RUSTUP_DIST_ROOT` has been set to `{url}`"))
-                .as_ref()
-                .map(|root| root.trim_end_matches("/dist"))
-                .unwrap_or(dist::DEFAULT_DIST_SERVER)
-                .to_owned()
-        }
-    })
+        // For backwards compatibility
+        else if let Some(root) = non_empty_env_var("RUSTUP_DIST_ROOT", process)? {
+            trace!("`RUSTUP_DIST_ROOT` has been set to `{root}`");
+            root.trim_end_matches("/dist").to_owned()
```

Comment:
```suggestion
            root.trim_end_matches("/dist").to_owned()
```

`.trim_end_matches()` repeatedly removes the suffix. I guess that's not what the original meant?

Reply:
Hm, oh yes that's wrong. I guess it's unlikely to make much difference in practice (`/dist/dist` would be unusual) but that should be fixed.

Reply:
I mean, assuming there's no backwards compatibility reason to keep it behaving the same (and I don't think there is)?

Reply:
@ChrisDenton Looking at the original commit (https://github.com/rust-lang/rustup/commit/b7e2a6ebd4990eb63aa057ee0e001479119d17ee) I believe this was just a mistake in the original implementation.

Reply:
I don't believe many are using it, since that commit was from nearly 10 years ago. I just checked and it seems to be the only really problematic use of `.trim_*_matches()` in the business logic.